runtime/compiler/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -11045,7 +11045,7 @@ J9::Z::TreeEvaluator::VMnewEvaluator(TR::Node * node, TR::CodeGenerator * cg)
 
                TR::Register *offsetReg = cg->allocateRegister();
                iCursor = generateRRInstruction(cg, TR::InstOpCode::getXORRegOpCode(), node, offsetReg, offsetReg, iCursor);
-               iCursor = generateRILInstruction(cg, TR::InstOpCode::getCmpImmOpCode(), node, enumReg, 0, iCursor);
+               iCursor = generateRILInstruction(cg, TR::InstOpCode::CFI, node, enumReg, 0, iCursor);
```

Comment:
Please add comment (As the enumReg should only have length in the 32-63 bit (Where depending on the case, 0-31 bits may contain the Class register or null). Also can we use `CHI` - we would save two bytes of instruction with that.

Reply:
Yup, CHI should work as well. I have updated the PR.